metaflow/runner/nbdeploy.py

```diff
@@ -0,0 +1,128 @@
+import os
+import tempfile
+from typing import Dict, Optional
+
+from metaflow import Deployer
+from metaflow.runner.utils import get_current_cell, format_flowfile
+
+DEFAULT_DIR = tempfile.gettempdir()
+
+
+class NBDeployerInitializationError(Exception):
+    """Custom exception for errors during NBDeployer initialization."""
+
+    pass
+
+
+class NBDeployer(object):
+    """
+    A  wrapper over `Deployer` for deploying flows defined in a Jupyter
+    notebook cell.
+
+    Instantiate this class on the last line of a notebook cell where
+    a `flow` is defined. In contrast to `Deployer`, this class is not
+    meant to be used in a context manager.
+
+    ```python
+    deployer = NBDeployer(FlowName)
+    ar = deployer.argo_workflows(name="madhur")
+    ar_obj = ar.create()
+    result = ar_obj.trigger(alpha=300)
+    print(result.status)
+    print(result.run)
+    result.terminate()
+    ```
+
+    Parameters
+    ----------
+    flow : FlowSpec
+        Flow defined in the same cell
+    show_output : bool, default True
+        Show the 'stdout' and 'stderr' to the console by default,
+    profile : Optional[str], default None
+        Metaflow profile to use to deploy this run. If not specified, the default
+        profile is used (or the one already set using `METAFLOW_PROFILE`)
+    env : Optional[Dict], default None
+        Additional environment variables to set. This overrides the
+        environment set for this process.
+    base_dir : Optional[str], default None
+        The directory to run the subprocess in; if not specified, a temporary
+        directory is used.
+    **kwargs : Any
+        Additional arguments that you would pass to `python myflow.py` i.e. options
+        listed in `python myflow.py --help`
+
+    """
+
+    def __init__(
+        self,
+        flow,
+        show_output: bool = True,
+        profile: Optional[str] = None,
+        env: Optional[Dict] = None,
+        base_dir: str = DEFAULT_DIR,
+        **kwargs,
+    ):
+        try:
+            from IPython import get_ipython
+
+            ipython = get_ipython()
+        except ModuleNotFoundError:
+            raise NBDeployerInitializationError(
+                "'NBDeployer' requires an interactive Python environment (such as Jupyter)"
+            )
+
+        self.cell = get_current_cell(ipython)
+        self.flow = flow
+        self.show_output = show_output
+        self.profile = profile
+        self.env = env
+        self.cwd = base_dir
+        self.top_level_kwargs = kwargs
+
+        self.env_vars = os.environ.copy()
+        self.env_vars.update(env or {})
+        self.env_vars.update({"JPY_PARENT_PID": ""})
```

Comment:
What is this? Can you add a comment explaining this line?

Reply:
This was present in original NBRunner as well...
I don't know what this does -- I remember ville doing it and thus I just borrowed it from there...